### PR TITLE
[SPARK-33040][R][ML] Add SparkR wrapper for vector_to_array

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

Add SparkR wrapper for `o.a.s.ml.functions.vector_to_array`

### Why are the changes needed?

- Currently ML vectors, including predictions, are almost inaccessible to R users. That's is a serious loss of functionality.
- Feature parity.

### Does this PR introduce _any_ user-facing change?

Yes, new R function is added.

### How was this patch tested?

- New unit tests.
- Manual verification.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @WeichenXu123 and @mengxr.

**Comment by zero323** [comment]
Retest this please.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: R/pkg/tests/fulltests/test_sparkSQL.R [hunk 1]
```diff
@@ -1424,7 +1424,8 @@ test_that("column functions", {
     date_trunc("quarter", c) + current_date() + current_timestamp()
   c25 <- overlay(c1, c2, c3, c3) + overlay(c1, c2, c3) + overlay(c1, c2, 1) +
     overlay(c1, c2, 3, 4)
-  c26 <- timestamp_seconds(c1)
+  c26 <- timestamp_seconds(c1) + vector_to_array(c) +
+    vector_to_array(c, "float32") + vector_to_array(c, "float64")
```
Comment (HyukjinKwon) [inline R/pkg/tests/fulltests/test_sparkSQL.R:1428]: quick question @zero323. why did we don't do the default argument here?

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -4458,3 +4469,25 @@ setMethod("timestamp_seconds",
...
+#' a column of dense arrays.
+#'
+#' @param dtype The data type of the output array. Valid values: "float64" or "float32".
+#'
+#' @rdname column_ml_functions
+#' @aliases vector_to_array vector_to_array,Column-method
+#' @note vector_to_array since 3.1.0
+setMethod("vector_to_array",
+          signature(x = "Column"),
+          function(x, dtype = c("float32", "float64")) {
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R:4484]: @zero323, sorry I missed this. Should we switch the order of the vector here to make the default as `float64` like Python and Scala side?

File: R/pkg/R/functions.R [hunk 3]
```diff
@@ -4458,3 +4469,25 @@ setMethod("timestamp_seconds",
...
+
+#' @details
+#' \code{vector_to_array} Converts a column of MLlib sparse/dense vectors into
+#' a column of dense arrays.
+#'
+#' @param dtype The data type of the output array. Valid values: "float64" or "float32".
+#'
+#' @rdname column_ml_functions
+#' @aliases vector_to_array vector_to_array,Column-method
+#' @note vector_to_array since 3.1.0
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R:4481]: Can we add an example ( `dontrun`) while we're making a followup to address https://github.com/apache/spark/pull/29917/files#r499340542?

File: R/pkg/R/functions.R [hunk 4]
```diff
@@ -345,6 +345,17 @@ NULL
...
 
+#' ML functions for Column operations
+#'
+#' ML functions defined for \code{Column}.
+#'
+#' @param x Column to compute on.
+#' @param ... additional argument(s).
+#' @name column_ml_functions
+#' @rdname column_ml_functions
+#' @family ml functions
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R:356]: Can we add an example ( `dontrun`) while we're making a followup to address https://github.com/apache/spark/pull/29917/files#r499340542?

**Pull request closed without merging**